view/components/main.html

```diff
@@ -38,79 +16,101 @@
+                    class="chat-screen-messages-jarvis-bot" class="clearfix">
+                    <div class="message-style">
+                        {{message.message}} at {{message.result.city}}
+                        <span>
+                            <sub class="show-user-creds">
+                                {{message.sender}}
+                                {{message.time}}
+                            </sub>
+                        </span>
+                    </div>
+                    <div id="queried-data" data-ng-if="message.result.length !== 0">
+                        <br>
+                        <div>
+                            <span class="head-query-data">Currently {{message.result.temperature}} &deg;C</span><br>
+                            <span class="head-query-data">Feels like {{message.result.feels_like}} &deg;C</span><br>
+                            <span class="head-query-data">Humidity being {{message.result.humidity}}</span><br>
+                            <span class="head-query-data">Visibility {{message.result.visibility}}</span><br>
+                            <span class="head-query-data">Dew Point {{message.result.dew_point}}</span><br>
+                            <span class="head-query-data">At time {{message.result.time}}</span>
+                        </div>
                     </div>
                 </div>
+                <br>
             </div>
-            <br>
         </div>
-    </div>
-    <!-- message field -->
-    <div id="message-bar">
-        <div id="message-input-parent">
-            <input type="text" name="" id="message-input" data-ng-model="message" data-ng-change="removeMessage()">
-            <button id="message-bar-send" class="btn btn-danger" data-ng-click="addMessagesToStack()">
-                Send
-            </button>
-            <!-- banner -->
-            <div id="banner-listen" data-ng-show="showLoaderListening">
-                <span>Listening ... </span>
-                <img src="../assets/icons/load.gif" alt="listening to you. Please keep speaking.." id="loader" srcset="">
+        <!-- message field -->
+        <div id="message-bar">
+            <div id="message-input-parent">
+                <input type="text" name="" id="message-input" data-ng-model="message" placeholder="Type a message">
+                <button id="message-bar-send" class="btn btn-danger" data-ng-click="addMessagesToStack()">
+                    Send
+                </button>
+                <!-- banner -->
+                <div id="banner-listen" data-ng-show="showLoaderListening">
+                    <span>Listening ... </span>
+                    <img src="../assets/icons/load.gif" alt="listening to you. Please keep speaking.." id="loader" srcset="">
+                </div>
+                <img src="../assets/icons/mic.png" alt="Click to Start/Stop speaking" id="button"
```

Comment:
could you please confirm if this functionality is working in the newer changes you have pushed?

Reply:
The button works, the gif loads when I click the button and vanishes when I click on it again.